### PR TITLE
Plugin: tester

### PR DESCRIPTION
**Pull request by sormuras**
## plugin `tester`

See work-in-progress output: https://travis-ci.org/forax/pro/builds/225994278

## ~todo~ done

- [x] Computer exit code and return it from `TesterPlugin#execute`
- [x] Make plugin `tester` configurable -- pass [options](http://junit.org/junit5/docs/current/user-guide/#running-tests-console-launcher-options) as raw arguments to wrapped JUnit Platform ConsoleLauncher
- [x] Make test classes compile to "local" target folder
- [x] Add "core pro" dependency to "junit.jupitert.api" and...
- [x] ...implement "core pro" example tests below `./src/test/java`

### CODE REVIEW

**Comment by sormuras** [comment]
@forax Is directory `./deps` the right place to put test scope libraries? See https://github.com/forax/pro/pull/27/commits/3079fdde5684bac47cfd085fcd0304e5beed5f45

**Comment by forax** [comment]
yes,
source dependencies and test dependencies are both stored in deps,
given that tests have a different module-info than sources, the module system will not mix then when compiling or running.

**Comment by sormuras** [comment]
> [...] the module system will not mix then when compiling or running.

But my _mind system_ sometimes does... ;)

**Comment by sormuras** [comment]
I'll address the last two open tasks today. I'm not sure, whether _Move "test.pro" script to Bootstrap_ is a ... good move. Maybe an extra build script for tests is clearer separation of concerns. What do you think?

File: plugins/tester/src/main/java/com.github.forax.pro.plugin.tester/com/github/forax/pro/plugin/tester/TesterPlugin.java [hunk 1]
```diff
@@ -77,28 +77,30 @@ int executeConsoleLauncher(String[] options) throws IOException {
...
-    Consumer<String> addClassPath = path -> {
+    Consumer<Path> addClassPath = path -> {
       options.add("--classpath");
-      options.add(path);
+      options.add(path.toString());
     };
-    findDirectoriesAt("target/test/exploded").forEach(addClassPath);
-    findDirectoriesAt("plugins")
-        .forEach(plugin -> findDirectoriesAt(plugin + "/target/test/exploded")
+    // TODO    javaModuleExplodedTestPaths = config.get("???.javaModuleExplodedTestPath", List.class).get();
```
Comment (sormuras) [inline plugins/tester/src/main/java/com.github.forax.pro.plugin.tester/com/github/forax/pro/plugin/tester/TesterPlugin.java]: @forax How do I extract the list of exploded test paths from the config object?

**Comment by forax** [comment]
You need to define a Facade interface to extract values from the "convention",
like https://github.com/forax/pro/blob/master/src/main/java/com.github.forax.pro.plugin.compiler/com/github/forax/pro/plugin/compiler/ConventionFacade.java
so you can extract it in execute.

Technically, each plugin defines an intermediary values that allows to override moduleExplodedPath but only for this plugin, like this:
https://github.com/forax/pro/blob/master/src/main/java/com.github.forax.pro.plugin.compiler/com/github/forax/pro/plugin/compiler/CompilerPlugin.java#L53
and
https://github.com/forax/pro/blob/master/src/main/java/com.github.forax.pro.plugin.compiler/com/github/forax/pro/plugin/compiler/CompilerPlugin.java#L111

**Comment by sormuras** [comment]
Like done in https://github.com/forax/pro/pull/27/commits/4db5328bcf11d02938eeb05175637237dfc976a7 ?

**Comment by forax** [comment]
yes !

**Comment by sormuras** [comment]
With the questionable task "Move `test.pro` script to `Bootstrap`" removed from the todo list, this PR feels like ... done. From my point of view. Anything _big_ left open, @forax ?

Cleanup, polishing and making `tester` really configurable is up to you. ✨ (-:

**Pull request merged**